src/awkward/operations/ak_from_buffers.py

```diff
@@ -300,8 +323,13 @@ def _reconstitute(form, length, container, getkey, backend, byteorder, simplify)
             count=length,
             byteorder=byteorder,
         )
-        reduced_stops = stops[starts != stops]
-        next_length = 0 if len(starts) == 0 else backend.index_nplike.max(reduced_stops)
+        if isinstance(stops, PlaceholderArray):
+            next_length = unknown_length
+        else:
+            reduced_stops = stops[starts != stops]
+            next_length = (
+                0 if len(starts) == 0 else backend.index_nplike.max(reduced_stops)
+            )
```

Comment:
This idiom is repeated often, which makes it look like it wants to be a helper function, but if it was a helper function, it would have to take a function of "what to do if not `PlaceholderArray`" as an argument, and that kind of callback-hell would be worse than this repetition. So I'm in favor of what you've done here.